src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -478,32 +482,43 @@ private void InitializePackagesFolderInfo(string packagesFolderPath, bool exclud
                 throw new ArgumentNullException(nameof(secondarySources));
             }
 
-            if (nuGetProject is INuGetIntegratedProject)
-            {
-                // project.json based projects are handled here
-                return await PreviewUpdatePackagesForBuildIntegratedAsync(
-                    packageId,
-                    packageIdentities,
-                    nuGetProject,
-                    resolutionContext,
-                    nuGetProjectContext,
-                    primarySources,
-                    secondarySources,
-                    token);
-            }
-            else
+            var tasks = new List<Task<IEnumerable<NuGetProjectAction>>>();
+
+            foreach (var project in nuGetProjects)
             {
-                // otherwise classic style packages.config style projects are handled here
-                return await PreviewUpdatePackagesForClassicAsync(
-                    packageId,
-                    packageIdentities,
-                    nuGetProject,
-                    resolutionContext,
-                    nuGetProjectContext,
-                    primarySources,
-                    secondarySources,
-                    token);
+                if (project is INuGetIntegratedProject)
+                {
+                    // project.json based projects are handled here
+                    tasks.Add(Task.Run(async ()
+                        => await PreviewUpdatePackagesForBuildIntegratedAsync(
+                            packageId,
+                            packageIdentities,
+                            project,
+                            resolutionContext,
+                            nuGetProjectContext,
+                            primarySources,
+                            secondarySources,
+                            token)));
+                }
+                else
+                {
+                    // otherwise classic style packages.config style projects are handled here
+                    tasks.Add(Task.Run(async ()
```

Comment:
AFAIK resolution context is being shared among all these requests so there shouldn't be duplicate calls and when i tested it with PC based projects... then i didn't see any issue. But I'll double check it. And yes I'll add throttling like while i add for uwp projects.